api/notd/block_processor.py

```diff
@@ -99,18 +102,35 @@ async def _get_retrieved_events(self, blockNumber: int) -> Dict[str, List[Retrie
             transactionHashEventMap[retrievedEvent.transactionHash].append(retrievedEvent)
         return transactionHashEventMap
 
+    async def _get_transaction_weth_values(self, blockNumber: int) -> List[Dict[str, Tuple(str,int)]]:
+        transactionWethValues = []
+        erc20events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc20TransferEventSignatureHash])
+        for event in erc20events:
+            transactionWethValues += await self._process_erc20_event(event=dict(event))
+        return transactionWethValues
+
     async def process_block(self, blockNumber: int) -> ProcessedBlock:
         # NOTE(krishan711): some blocks are too large to be retrieved from the AWS hosted node e.g. #14222802
         # for these, we can use infura specifically but if this problem gets too big find a better solution
         blockData = await self.ethClient.get_block(blockNumber=blockNumber, shouldHydrateTransactions=True)
         transactionHashEventMap = await self._get_retrieved_events(blockNumber=blockNumber)
+        transactionWethValues = await self._get_transaction_weth_values(blockNumber=blockNumber)
         retrievedTokenTransfers: List[RetrievedTokenTransfer] = []
         for transaction in blockData['transactions']:
-            retrievedTokenTransfers += await self.process_transaction(transaction=transaction, retrievedEvents=transactionHashEventMap[transaction['hash'].hex()])
+            retrievedTokenTransfers += await self.process_transaction(transaction=transaction, retrievedEvents=transactionHashEventMap[transaction['hash'].hex()], transactionWethValues=transactionWethValues)
         blockHash = blockData['hash'].hex()
         blockDate = datetime.datetime.utcfromtimestamp(blockData['timestamp'])
         return ProcessedBlock(blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate, retrievedTokenTransfers=retrievedTokenTransfers)
 
+    async def _process_erc20_event(self, event: LogReceipt) -> List[Dict[str, Tuple(str,int)]]:
+        # NOTE(Femi-Ogunkola): Limit to just weth
+        if len(event['topics']) == 3 and event['address'] == WRAPPED_ETHER_ADDRESS:
+            transactionHash = event['transactionHash'].hex()
+            toAddress = chain_util.normalize_address(event['topics'][1].hex())
+            wethValue = int(event["data"], 16)
+            return [{transactionHash: (toAddress, wethValue)}]
```

Comment:
why are you returning a list here, this function seems quite wierd to me

Reply:
in fact this function can just be inlined, its only used in one place and would make the other function much easier to understand